fix(io): add optional 'quotechar' parameter to import_from_csv_folder()

gulbord opened the pull request:
## Changes proposed in this Pull Request

Hello! I wanted to propose this small edit (`io` is the only module involved) because I faced an issue in importing a network (the one from [1, 2]) with `Network.import_from_csv_folder`.

I just added a `quotechar: str = '"'` argument to the `import_from_csv_folder` method to handle single-quoted linestrings in the geometry column, like in the case I mentioned. The parameter is passed down to `ImporterCSV` and the `pandas.read_csv` calls in its methods. The docstring is updated.

[1] B. Xiong, D. Fioriti, F. Neumann, I. Riepin, and T. Brown, ‘Modelling the High-Voltage Grid Using Open Data for Europe and Beyond’, *arXiv [physics.soc-ph]*. 2024.
[2] B. Xiong, D. Fioriti, F. Neumann, I. Riepine T. Brown, ‘Prebuilt Electricity Network for PyPSA-Eur based on OpenStreetMap Data’. Zenodo, nov. 13, 2024. doi: [10.5281/zenodo.14144752](https://doi.org/10.5281/zenodo.14144752).

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [x] Unit tests for new features were added (if applicable).
- [x] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.
- [x] I consent to the release of this PR's code under the MIT license.

lkstrp reviewed (commented):
@gulbord Thanks for the contribution! 

The `quotechar` argument is useful. But why not add it to the export function as well?

Also, could you maybe add a little test for the argument and add a release note?

Looking at the code in general, this could be implemented and integrated better with pandas. But I guess this is outside the scope of this PR and more of an `io` overhaul.

gulbord commented:
Thanks for the review! I added the parameter to the exporter and updated tests and documentation as suggested, let me know if there are any problems I might have overlooked.
